pkg/crd/known_types.go

```diff
@@ -73,6 +73,13 @@ var KnownPackages = map[string]PackageOverride{
 		}
 		// No point in calling AddPackage, this is the sole inhabitant
 	},
+
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1": func(p *Parser, pkg *loader.Package) {
+		p.Schemata[TypeIdent{Name: "JSON", Package: pkg}] = apiext.JSONSchemaProps{
+			Type: "string",
```

Comment:
I'm not 100% sure if the type here should be `string` or something else - the existing Golang code doesn't actually specify.

cc @sttts any ideas?

Reply:
type string is wrong. This is arbitrary JSON, like RawExtension but without Type/ObjectMeta.

Reply:
After some testing, I've updated this to `object`. I think this is correct, although it *may* not accept data which is a string whereas it probably should given JSON is meant to be completely unstructured.

What's the proper way to set this? It looks like the Kubernetes APIServer exposes an `object` with a field named `Raw` of type `string` in the OpenAPI schema for RawExtension which I think would *also* be incorrect?

Reply:
Also object is wrong. This is pure JSON, any type. In other words, no type should be set at all.

But you have to set `x-kubernetes-preserve-unknown-fields: true`.

Reply:
```
apiext.JSONSchemaProps{
  XPreserveUnknownFields: true,
}
```

Reply:
👍 thanks, so it seems like this requires #276 then!

Reply:
yes